be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/MessageDecoder.scala

```diff
@@ -2,15 +2,31 @@ package ch.epfl.pop.pubsub.graph
 
 import akka.NotUsed
 import akka.stream.scaladsl.Flow
-import ch.epfl.pop.model.network.{JsonRpcRequest, MethodType}
+import ch.epfl.pop.model.network.{JsonRpcRequest, JsonRpcResponse}
+import ch.epfl.pop.jsonNew.HighLevelProtocol._
+import spray.json._
+
+import scala.util.{Success, Try}
 
 object MessageDecoder {
 
-  // Message or TextMessage
   val jsonRpcParser: Flow[Either[JsonString, PipelineError], GraphMessage, NotUsed] = Flow[Either[JsonString, PipelineError]].map {
```

Comment:
IIRC, `Either[PipelineError, JsonString]` would be the convention

Reply:
Correct! I can change this one now since it's only used in this PR :)

Reply:
EDIT : nvm :/ Will have to wait ffor the merge as well